projects/storefrontlib/cms-components/misc/site-context-selector/site-context-component.service.ts

```diff
@@ -71,17 +73,28 @@ export class SiteContextComponentService {
     context?: SiteContextType
   ): Observable<SiteContext<any>> {
     return this.getContext(context).pipe(
-      map((ctx: string) => (ctx ? this.getInjectedService(ctx) : undefined)),
-      filter((s) => !!s)
+      map((ctx: string | undefined) =>
+        ctx ? this.getInjectedService(ctx) : undefined
+      ),
+      filter(isNotUndefined)
     );
   }
 
-  protected getContext(context?: SiteContextType): Observable<string> {
+  protected getContext(
+    context?: SiteContextType
+  ): Observable<string | undefined> {
     if (context) {
-      return of(context);
+      if (context === SiteContextType.CURRENCY) {
+        return of(CURRENCY_CONTEXT_ID);
+      } else if (context === SiteContextType.LANGUAGE) {
+        return of(LANGUAGE_CONTEXT_ID);
+      } else {
```

Comment:
Was that new logic required for strict typing?

Reply:
yes, before it returns `of(context)`, which is `Observable<SiteContextType>`. This function needs return `Observable<string>`.